bundles/AdminBundle/Resources/public/js/pimcore/elementservice.js

```diff
@@ -119,10 +119,12 @@ pimcore.elementservice.deleteElementFromServer = function (r, options, button) {
         var successHandler = options["success"];
         var elementType = options.elementType;
         var id = options.id;
+        const preDeleteEventName = 'preDelete' + elementType.charAt(0).toUpperCase() + elementType.slice(1);
 
         let ids = Ext.isString(id) ? id.split(',') : [id];
         ids.forEach(function (elementId) {
             pimcore.helpers.addTreeNodeLoadingIndicator(elementType, elementId);
+            pimcore.plugin.broker.fireEvent(preDeleteEventName, elementId);
```

Comment:
```suggestion
           try {
                pimcore.plugin.broker.fireEvent(preDeleteEventName, elementId);
            } catch (e) {
                pimcore.helpers.showPrettyError('asset', t("error"), t("delete_failed"), e.message);
                return false;
            }

            pimcore.helpers.addTreeNodeLoadingIndicator(elementType, elementId);
```

something like this @fashxp ?